src/daypicker/DayPickerTop.js

```diff
@@ -1,25 +1,32 @@
 import React, {PropTypes} from 'react';
 import partial from 'lodash/function/partial';
+import capitalize from 'lodash/string/capitalize';
 import DateUtils from '../utils/DateUtils.js';
 import PickerTop from '../PickerTop';
 
-const DayPickerTop = React.createClass({
+export default React.createClass({
+
+  displayName: 'DayPickerTop',
 
   propTypes: {
-    visibleDate: PropTypes.any.isRequired,
+    initialVisibleDate: PropTypes.any.isRequired,
     onChangeVisibleDate: PropTypes.func.isRequired,
     onChangeMode: PropTypes.func.isRequired,
     fixedMode: PropTypes.bool
   },
 
-  changeMonth(month) {
-    this.props.visibleDate.month(month);
-    this.props.onChangeVisibleDate(this.props.visibleDate);
+  getInitialState() {
+    return { visibleDate: this.props.initialVisibleDate.clone() };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ visibleDate: nextProps.initialVisibleDate });
```

Comment:
Why do we need internal state if `this.state.visibleDate` is always equal to `this.props.initialVisibleDate`?
Stateless components are easier to test, I suggest we use only `this.props.visibleDate` and delegate the state change to its father